boms/geode-all-bom/src/test/resources/expected-pom.xml

```diff
@@ -1045,6 +1039,12 @@
         <version>${version}</version>
         <scope>compile</scope>
       </dependency>
+      <dependency>
+        <groupId>org.apache.geode</groupId>
+        <artifactId>geode-deployment-legacy</artifactId>
+        <version>${version}</version>
+        <scope>compile</scope>
```

Comment:
I see that we decided to leave this module as non-optional. Can you explain the reasoning there?

Reply:
The reason why it was decided to mark it as "required" in this PR is because this PR is the pre-cursor to GEODE-8705. In GEODE-8705 this would become "optional" but for now this project is required at runtime.

Reply:
ok.